.github/workflows/ci-pr.yml

```diff
@@ -0,0 +1,138 @@
+name: Deploy PR previews
+concurrency: preview-${{ github.ref }}
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - closed
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+      pull-requests: write
+      pages: write
+      id-token: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "16.19.0"
+
+      - name: Install Dependencies
+        run: yarn
+
+      - name: Filter the packages
```

Comment:
Filter packages?

Reply:
Yes

Reply:
it checks that which package of monorepo has changes if there is change in react monorepo then it will deploy only react one, if there is change in ui-elements it will deploy only ui-elements and so on..